### DISCUSSION TITLE
Brew uses a system path as if it were a user path

### DISCUSSION THREAD

**Discussion by alejandro-colomar** [Homebrew/discussions#3432] [Getting started]
`/usr/local/var/log` is a system path, owned by root.  `brew` expects it to be owned by `$(whoami)`.

I installed some software from source with the usual prefix of `/usr/local`, which created the previously-non-existing `/usr/local/var/log` directory, as part of `sudo make install`.

Then, I needed to install `tree`(1) with `brew`, and so I run `brew install tree`.  No good.  It errored because it didn't have access to `/usr/local/var/log` (but it didn't exist before, so I guess it didn't need it before? why now?).

I don't think `brew` should expect that path to be property of my user, do you?

Maybe brew should create a subdir in it and ask for a root password to do so?  That way it could mess with it without requiring me to take ownership of a system path.

Maybe `brew` should be redesigned to default to `$HOME/bin` and similar paths?

**Comment by SMillerDev** [comment]
> /usr/local/var/log is a system path, owned by root. brew expects it to be owned by $(whoami).

> I don't think brew should expect that path to be property of my user, do you?

It's not a standard system path, and if you use `/usr/local` for homebrew it's not that weird that it would want to write there right? Same could be said for `/usr/local/bin` that could be created by a different user before you installed homebrew, but homebrew would need to write there anyway.

> Maybe brew should create a subdir in it and ask for a root password to do so?

No, https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

> Maybe brew should be redesigned to default to $HOME/bin and similar paths?

You could make a pull request, but seeing how you're the first person to encounter this problem I wouldn't expect such a major architecture change to gain much support.

**Reply by alejandro-colomar** [reply]
> > /usr/local/var/log is a system path, owned by root. brew expects it to be owned by $(whoami).
> 
> > I don't think brew should expect that path to be property of my user, do you?
> 
> It's not a standard system path, and if you use `/usr/local` for homebrew it's not that weird that it would want to write there right? Same could be said for `/usr/local/bin` that could be created by a different user before you installed homebrew, but homebrew would need to write there anyway.

Depending on what we consider standard.  It's, at least, standardised by GNU (not `/usr/local/var/log`, but `/usr/local/var`; but `log` is a standard directory within `var`):

<https://www.gnu.org/prep/standards/html_node/Directory-Variables.html>
<https://lists.gnu.org/archive/html/bug-make/2022-07/msg00019.html>

> 
> > Maybe brew should create a subdir in it and ask for a root password to do so?
> 
> No, https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

Okay

> 
> > Maybe brew should be redesigned to default to $HOME/bin and similar paths?
> 
> You could make a pull request, but seeing how you're the first person to encounter this problem I wouldn't expect such a major architecture change to gain much support.

I guess...

**Comment by carlocab** [comment]
> Maybe `brew` should be redesigned to default to `$HOME/bin` and similar paths?

This doesn't really require changes in `brew` -- `brew` can easily make this work.

What this requires changes in are the hundreds (possibly thousands) of packages that won't work for you because `$HOME` on our build machine isn't the same as `$HOME` on your machine.

**Reply by alejandro-colomar** [reply]
Aaand, that's why it's a system install, I think we agree.  But, brew pretends to own (part of) the system from a non-system user.  That's where I think the design is flawed.

What if brew asked for root privileges to create symlinks in system dirs as root, and installed in a user path (which can be a .../Cellar or .../Homebrew subdir within standard directories)?  That's more or less what it already does, but without owning the symlinks also from the user.
